### PR TITLE
Switch project from formatter-maven-plugin to spotless-maven-plugin

### PR DESCRIPTION
**Pull request by Donnerbart**
See https://github.com/operator-framework/josdk-webhooks/pull/284#discussion_r1848515774

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -171,24 +170,33 @@
                     <version>${maven-install-plugin.version}</version>
                 </plugin>
                 <plugin>
-                    <groupId>net.revelc.code.formatter</groupId>
-                    <artifactId>formatter-maven-plugin</artifactId>
-                    <version>${formatter-maven-plugin.version}</version>
+                    <groupId>com.diffplug.spotless</groupId>
+                    <artifactId>spotless-maven-plugin</artifactId>
+                    <version>${spotless-maven-plugin.version}</version>
                     <configuration>
```
Comment (Donnerbart) [inline pom.xml:176]: Based on https://github.com/fabric8io/kubernetes-client/blob/main/pom.xml#L1102

File: pom.xml [hunk 2]
```diff
@@ -171,24 +170,33 @@
...
+                            <eclipse>
+                                <file>ide-config/eclipse-format.xml</file>
+                            </eclipse>
+                            <importOrder>
+                                <file>ide-config/eclipse.importorder</file>
+                            </importOrder>
+                            <removeUnusedImports/>
+                        </java>
                     </configuration>
+                    <executions>
```
Comment (Donnerbart) [inline pom.xml:191]: Based on https://github.com/diffplug/spotless/tree/main/plugin-maven#binding-to-maven-phase, but somehow this doesn't work for me. Also without this configuration no `spotless:check` is executed when I run `mvn verify` (which should be the default).

Comment (csviri) [reply pom.xml:191]: see it here: https://github.com/operator-framework/java-operator-sdk/blob/0c20e6b6726a483dbdb2c116089f02bbe0dc5bf9/pom.xml#L325-L355

File: pom.xml [hunk 3]
```diff
@@ -171,24 +170,33 @@
...
-                    <version>${impsort-maven-plugin.version}</version>
-                    <configuration>
-                        <cachedir>.cache</cachedir>
-                        <groups>java.,javax.,org.,io.,com.</groups>
-                        <staticGroups>*</staticGroups>
-                        <staticAfter>true</staticAfter>
-                        <removeUnused>true</removeUnused>
+                        <java>
+                            <includes>
+                                <include>src/main/java/**/*.java</include>
```
Comment (csviri) [inline pom.xml:179]: not sure we need these explicitly

File: pom.xml (hunk 2 above)
Comment (Donnerbart) [reply pom.xml]: I've aligned the config with `java-operator-sdk`, also kept the XML files in the `contributing` folder.

**Review by csviri — APPROVED** [review]
LGTM, thank you!

**Pull request merged**